### PR TITLE
Add samples and sample validation ci

### PR DESCRIPTION
**Pull request by YunchuWang**
# Summary

## What changed?
- Added 7 new runnable sample projects under `examples/azure-managed/` demonstrating all major SDK features:
  - **hello-orchestrations** — Activity chaining, fan-out/fan-in (`whenAll`), sub-orchestrations, `whenAny`, deterministic GUID (`newGuid`)
  - **retry-and-error-handling** — `RetryPolicy`, `handleFailure` predicate, `AsyncRetryHandler`, sub-orchestration retry, `raiseIfFailed()`
  - **human-interaction** — External events, timers, `whenAny` race pattern, `sendEvent`, `setCustomStatus`
  - **lifecycle-management** — Terminate (recursive), suspend/resume, continue-as-new, restart, purge, tags
  - **query-and-history** — `getAllInstances` with pagination, `listInstanceIds`, `getOrchestrationHistory`, typed history events
  - **versioning** — `VersionMatchStrategy`, `VersionFailureStrategy`, `ctx.version`, `ctx.compareVersionTo()`
  - **unit-testing** — `InMemoryOrchestrationBackend`, `TestOrchestrationClient`/`TestOrchestrationWorker`, `ReplaySafeLogger`, `NoOpLogger` (no emulator required)
- Each sample includes `sample.json` (metadata/discovery marker), `index.ts`, and `README.md`
- Added `examples/azure-managed/sample.json` marker for the existing root-level distributed-tracing example
- Added CI pipeline `.github/workflows/validate-samples.yaml` with auto-discovery, Docker service containers (DTS emulator), and matrix strategy
- Replaced `examples/azure-managed/README.md` with a comprehensive Samples Index including feature coverage map (40+ features → samples), quick-start instructions, and troubleshooting — existing distributed-tracing documentation preserved below the index

## Why is this change needed?
- The SDK exports 40+ public features (orchestration patterns, retry policies, lifecycle management, versioning, querying, in-memory testing, etc.) but had only 2 existing examples (hello-world and distributed-tracing), leaving most functionality undocumented by runnable code
- New contributors and users had no way to discover or learn advanced features without reading source code and e2e tests
- No CI validation existed for examples, so samples could silently break as the SDK evolves

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `CHANGELOG.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [ ] All required tests have been added/updated (unit tests, E2E tests)
  - Note: Samples are validated by the new `validate-samples.yaml` CI workflow; no unit/E2E test changes needed since these are documentation/example files
- [x] Breaking change?
  - No breaking changes. All additions are new files; no existing code was modified except:
    - `examples/azure-managed/README.md` (documentation only — prepended Samples Index, preserved existing content)
    - `test/e2e-azuremanaged/retry-handler.spec.ts` (removed 5 stale `eslint-disable` directives via `--fix`)

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot (Claude Opus 4.6, agent mode in VS Code)
- AI-assisted areas/files:
  - All 7 sample directories (`index.ts`, `README.md`, `sample.json`)
  - `.github/workflows/validate-samples.yaml`
  - `examples/azure-managed/README.md` (Samples Index)
- What you changed after AI output:
  - Reviewed and verified all API references against actual SDK exports in `packages/durabletask-js/src/index.ts` and `packages/durabletask-js-azuremanaged/src/index.ts`
  - Fixed 4 lint errors (missing `yield` in generators, unused imports/variables)
  - Auto-fixed 5 stale eslint-disable directives in `retry-handler.spec.ts`

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed — `npm run lint` passes with 0 errors, 0 warnings

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): Windows, Node.js ≥22, DTS emulator (Docker)
- Steps + observed results:
  1. `npm install && npm run build` — SDK builds successfully
  2. `npm run lint` — 0 errors, 0 warnings
  3. `unit-testing` sample runs without emulator via `InMemoryOrchestrationBackend`
- Evidence (optional): N/A

---

# Notes for reviewers
- Samples are grouped by **concern area** (core patterns, fault tolerance, event-driven, lifecycle, observability, versioning, testing) so related features are demonstrated together. Each sample is self-contained and can be read in under 5 minutes.
- The `unit-testing` sample is the only one with `"requiresEmulator": false` in its `sample.json` — CI runs it without Docker.
- The CI pipeline auto-discovers samples via `sample.json` marker files, so adding new samples requires no workflow changes.
- The feature coverage map in the README ensures every exported public API symbol maps to at least one sample.
- Files changed: **22 new files**, **1 modified** (`examples/azure-managed/README.md`), plus lint auto-fix in `retry-handler.spec.ts`.

### CODE REVIEW

File: examples/azure-managed/hello-orchestrations/index.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  // winner is the task that completed first
+  return { winnerResult: winner.getResult() };
+};
+
+/** 5. Deterministic GUID — generates replay-safe unique IDs. */
+const guidOrchestrator: TOrchestrator = async function* (ctx: OrchestrationContext): any {
+  // Call an activity so the generator has at least one yield point
+  const label: string = yield ctx.callActivity(greet, "GUID demo");
+  const guid1 = ctx.newGuid();
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply examples/azure-managed/hello-orchestrations/index.ts]: done

File: examples/azure-managed/versioning/index.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+  // --- 1. Schedule with version ---
+  console.log("\n=== 1. Schedule Orchestration with Version ===");
+
+  // Client with a default version
+  const clientV1 = new DurableTaskAzureManagedClientBuilder()
+    .connectionString(connectionString)
+    .logger(logger)
+    .build();
+
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply examples/azure-managed/versioning/index.ts]: done
